datastore/longhorn.go

```diff
@@ -210,14 +214,13 @@ func (s *DataStore) GetSetting(sName types.SettingName) (*longhorn.Setting, erro
 		if !ErrorIsNotFound(err) {
 			return nil, err
 		}
-		resultRO = &longhorn.Setting{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: string(sName),
-			},
-			Setting: types.Setting{
-				Value: definition.Default,
-			},
+
+		defaultSettings, err := types.GetCustomizedDefaultSettings()
```

Comment:
Think about it more, if the setting was initialized correctly, we shouldn't need this automatically refill in the GetSetting().

Reply:
Yeah, InitSetting was added after GetSetting automatically fill the default value. We shouldn't need it now.

Reply:
If we want to remove this automatic refill, we need to make sure `GetSetting()` won't be invoked before `InitSettings()` complete. But both `InitSettings()`and `GetSetting()` are called after we start controllers. 

If we just want to avoid reading the file in `GetSetting()`, we can overwrite the built-in default setting with customized settings before starting controllers. I think it is easier.

Reply:
Agree, I think it's better this way.